src/selectors/expandThoughts.ts

```diff
@@ -100,22 +135,31 @@ function expandThoughts(state: State, path: Path | null, { depth = 0, returnCont
     (grandchildren.length !== 1 || !isURL(childValue(grandchildren[0], showContexts)))
 
   const childrenPinned = isOnlyChildNoUrl || isTable(state, context) || pinChildren(state, context) || publishPinChildren(state, context)
-    ? children
-    : children.filter(child => {
+    ? visibleChildren
+    : visibleChildren.filter(child => {
       const value = strip(childValue(child, showContexts))
-      return value[value.length - 1] === EXPAND_THOUGHT_CHAR || isPinned(child) === 'true'
+
+      const childPath = unroot([...path || [], { ...child, value }])
+
+      const childContext = pathToContext(childPath)
+
+      /** Check of the path is the ancestor of the expansion path. */
+      const isAncestor = () => isDescendant(childContext, expansionPathContext)
```

Comment:
I think the same `isAncestor` is called in the previous `childrenUnfiltered.filter`. Is it doing the same work twice?

Reply:
Yes. I need to fix this.